src/vs/platform/windows/electron-main/windowImpl.ts

```diff
@@ -595,6 +601,25 @@ export class CodeWindow extends BaseWindow implements ICodeWindow {
 		}
 		//#endregion
 
+		//#region JS Callstack Collector
+
+		let sampleInterval = parseInt(this.environmentMainService.args['unresponsive-sample-interval'] || '1000');
+		let samplePeriod = parseInt(this.environmentMainService.args['unresponsive-sample-period'] || '15000');
+		if (sampleInterval <= 0 || samplePeriod <= 0 || sampleInterval > samplePeriod) {
+			this.logService.warn(`Invalid unresponsive sample interval (${sampleInterval}ms) or period (${samplePeriod}ms), using defaults.`);
+			sampleInterval = 1000;
+			samplePeriod = 15000;
+		}
+
+		this.jsCallStackMap = new Map<string, number>();
+		this.jsCallStackEffectiveSampleCount = Math.round(sampleInterval / samplePeriod);
```

Comment:
Won't this be zero, and then cause div-by-zero below?

```js
> parseInt('1000') / parseInt('15000')
0.06666666666666667
> Math.round(parseInt('1000') / parseInt('15000'))
0
```

Reply:
@deepak1556 can you check?

Reply:
Oops, good catch. It should have been the other way.

```
this.jsCallStackEffectiveSampleCount = Math.round(samplePeriod / sampleInterval);
```